osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/periodicmeterreads/GetPeriodicMeterReadsCommandExecutorTest.java

```diff
@@ -110,43 +163,74 @@ public void testHappy() throws Exception {
+        when(this.dlmsHelper.getAndCheck(this.connectionManager, this.device, expectedDescription,
+                attributeAddress)).thenReturn(Collections.singletonList(getResult));
+
+        when(this.dlmsHelper.readDataObject(eq(getResult), any(String.class))).thenReturn(resultData);
+
         final CosemDateTimeDto cosemDateTime = mock(CosemDateTimeDto.class);
         final String expectedDateTimeDescription = String.format("Clock from %s buffer", periodType);
         when(this.dlmsHelper.readDateTime(data0, expectedDateTimeDescription)).thenReturn(cosemDateTime);
+
         final DateTime bufferedDateTime = DateTime.now();
         when(cosemDateTime.asDateTime()).thenReturn(bufferedDateTime);
 
-        // TODO: mock other calls
-
-        final PeriodicMeterReadsRequestDto request = new PeriodicMeterReadsRequestDto(periodType, new Date(from),
-                new Date(to));
+        final DlmsMeterValueDto meterValue1 = mock(DlmsMeterValueDto.class);
+        final DlmsMeterValueDto meterValue2 = mock(DlmsMeterValueDto.class);
+        when(this.dlmsHelper.getScaledMeterValue(data1, null, "positiveActiveEnergyTariff1")).thenReturn(meterValue1);
+        when(this.dlmsHelper.getScaledMeterValue(data4, null, "positiveActiveEnergyTariff2")).thenReturn(meterValue2);
 
         // CALL
         final PeriodicMeterReadsResponseDto result = this.executor.execute(this.connectionManager, device, request);
 
-        // VERIFY
-        verify(this.dlmsMessageListener).setDescription(
-                String.format("GetPeriodicMeterReads DAILY from %s until %s, retrieve attribute: {%s,%s,%s}",
-                        new DateTime(from), new DateTime(to), classId, address, id));
+        // VERIFY calls to mocks
+        verify(this.dlmsMessageListener).setDescription(String.format(
+                "GetPeriodicMeterReads DAILY from %s until %s, retrieve attribute: {%s,%s,%s}",
+                new DateTime(this.from), new DateTime(this.to), dlmsProfile.getClassId(), dlmsProfile.getObisCode(),
+                dlmsProfile.getDefaultAttributeId()));
 
-        verify(this.dlmsHelper).validateBufferedDateTime(same(bufferedDateTime), same(cosemDateTime),
+        verify(this.dlmsHelper, times(2)).validateBufferedDateTime(same(bufferedDateTime), same(cosemDateTime),
                 argThat(new DateTimeMatcher(from)), argThat(new DateTimeMatcher(to)));
 
+        verify(this.dlmsObjectConfigService).findDlmsObject(any(Protocol.class),
+                any(DlmsObjectType.class), any(Medium.class));
+
+        // ASSERT - the result should contain 2 values
         final List<PeriodicMeterReadsResponseItemDto> periodicMeterReads = result.getPeriodicMeterReads();
-        // TODO: assert contents of result
+
+        assertThat(periodicMeterReads.size()).isEqualTo(2);
+
+        periodicMeterReads.forEach(p -> {
+            assertThat(p.getLogTime()).isNotNull();
```

Comment:
En dan kan je hier controleren of de waardes goed gebruikt zijn:
            assertThat(p.getActiveEnergyImportTariffOne()).isEqualTo(meterValue1);
            assertThat(p.getActiveEnergyImportTariffTwo()).isEqualTo(meterValue2);
            assertThat(p.getActiveEnergyExportTariffOne()).isEqualTo(meterValue3);
            assertThat(p.getActiveEnergyExportTariffTwo()).isEqualTo(meterValue4);

Reply:
done